pkg/image/util/resolve.go

```diff
@@ -0,0 +1,78 @@
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/openshift/origin/pkg/client"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+// ResolveImagePullSpec resolves the provided source which can be "docker", "istag" or
+// "isimage" and returns the full Docker pull spec.
+func ResolveImagePullSpec(client *client.Client, source, name, defaultNamespace string) (string, error) {
+	// parse the namespace from the provided image
+	namespace, image := splitNamespaceAndImage(name)
+	if len(namespace) == 0 {
+		namespace = defaultNamespace
+	}
+
+	dockerImageReference := ""
+
+	if IsImageStreamTag(source) {
+		name, tag, ok := imageapi.SplitImageStreamTag(image)
+		if !ok {
+			return "", fmt.Errorf("invalid image stream tag %q, must be of the form [NAMESPACE/]NAME:TAG", name)
+		}
+		if resolved, err := client.ImageStreamTags(namespace).Get(name, tag); err != nil {
+			return "", fmt.Errorf("failed to get image stream tag %q: %v", name, err)
+		} else {
+			dockerImageReference = resolved.Image.DockerImageReference
+		}
+	}
+
+	if IsImageStreamImage(source) {
+		name, digest, ok := imageapi.SplitImageStreamImage(image)
+		if !ok {
+			return "", fmt.Errorf("invalid image stream image %q, must be of the form [NAMESPACE/]NAME@DIGEST", name)
+		}
+		if resolved, err := client.ImageStreamImages(namespace).Get(name, digest); err != nil {
+			return "", fmt.Errorf("failed to get image stream image %q: %v", name, err)
+		} else {
+			dockerImageReference = resolved.Image.DockerImageReference
+		}
+	}
+
+	if len(dockerImageReference) == 0 {
+		return "", fmt.Errorf("failed to resolve %q", name)
+	}
+
+	if reference, err := imageapi.ParseDockerImageReference(dockerImageReference); err != nil {
+		return "", err
+	} else {
+		return reference.String(), nil
+	}
```

Comment:
Change the if-else into:
```
reference, err := imageapi.ParseDockerImageReference(dockerImageReference)
if err != nil {
	return "", err
}
return reference.String(), nil
```
IIRC, either govet or golint will complain about missing return at the end of function.